numpy/core/tests/test_regression.py

```diff
@@ -1683,16 +1683,19 @@ def test_object_array_self_reference(self):
         # Object arrays with references to themselves can cause problems
         a = np.array(0, dtype=object)
         a[()] = a
-        assert_raises(TypeError, int, a)
-        assert_raises(TypeError, long, a)
-        assert_raises(TypeError, float, a)
-        assert_raises(TypeError, oct, a)
-        assert_raises(TypeError, hex, a)
+        assert_raises(RecursionError, int, a)
+        assert_raises(RecursionError, long, a)
+        assert_raises(RecursionError, float, a)
+        if sys.version_info.major == 2:
+            # in python 3, this falls back on operator.index, which fails on
+            # on dtype=object
+            assert_raises(RecursionError, oct, a)
+            assert_raises(RecursionError, hex, a)
 
         # Test the same for a circular reference.
         b = np.array(a, dtype=object)
         a[()] = b
-        assert_raises(TypeError, int, a)
+        assert_raises(RecursionError, int, a)
```

Comment:
[RecursionError](https://docs.python.org/3/library/exceptions.html?highlight=recursionerror#RecursionError) is a python3-ism, is there a trick for python 2?

Reply:
test_multiarray does:
```
        try:
            Error = RecursionError
        except NameError:
            Error = RuntimeError  # python < 3.5
        assert_raises(Error, bool, self_containing)  # previously stack overflow
```

Reply:
Yeah, we handle that elsewhere. Again, good catch!